src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -39,18 +39,21 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
         }
 
         $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}, config: array{vendor-dir: <string>}}|null $composer */
```

Comment:
Would it's be an idea to have the `json_decode()` for the composer mapped into a `ValueObject`? Instead of working with arrays?

Reply:
I think at the moment a VO just for this is possibly overkill; if we want to start looking at other `composer.json` keys in `config` in the future, something I'd certainly advocate.